### ISSUE TITLE
Disable the drop-down menu icons from the menu manager setup parameters

### ISSUE THREAD

**Issue by alisamii** [gantry/gantry5#701]
It would be great if we could add a simple checkbox to the menu settings that would allow you to disable the display of the drop-down menu arrow and cross icons.

As you can see from this screenshot, the icon takes up just enough space on my menu to wrap the menu on the last item and push it down to a new row.

This occurs on iPad landscape view.

It would be nice if we could set the menu not to display the arrow/cross icons.

![image](https://cloud.githubusercontent.com/assets/1498512/9218281/673ee3be-40d1-11e5-9351-40ec3666559a.jpg)

**Comment by mahagr** [comment]
You can just display one level of the menu (needs latest CI build). That will disable dropdown. You cannot just hide icon button only as it has different meaning than the item itself (for touch devices, including many laptops).

**Comment by alisamii** [comment]
Hi Matias,

Disabling the drop-down defeats the purpose of having the menu drop down, which I need.

The arrow/cross icon isn't a required element unless your site requires that users close the menu manually to proceed. It's functionality, furthermore, could be placed within the drop-down directly rather than in the Parent element.

I can override it in css but that's not the same as having the option to disable the display of the icon globally.

Best regards,

Ali

**Comment by w00fz** [comment]
There are two variations of that dropdown arrow:
 1. The item is a Parent AND a link (fat arrow shows up)
 2. The item is a Parent BUT a separator (simple arrow)

![variations](https://cloud.githubusercontent.com/assets/11734/9235781/cc3df2a4-40f5-11e5-9606-3c772fdcfe34.jpg)

We never encourage nor suggest to ever have a Parent Item being both a Parent and a Link to a page, this has been a huge issue in the past because what previously was handled on desktops easily via HOVER vs CLICK, with touch devices has become a challenge, since there is no HOVER concept but just tap.

With Gantry 5 we enhanced this by automatically handling a Parent Item that is also a Link to a page. By separating the text and the dropdown icon to their own entity, you can now perform both actions of click for the link or hover/click/tap for the dropdown.

Since the two behaviors are very different we had to make that dropdown icon standout, de facto that is the target for your hover and it needs to be big enough to be a target on touch devices. Limiting this to just an arrow with no padding nor backgrounds, will make it real hard to figure out that clicking/tapping on the text is not the same as clicking/tapping on the arrow.

I don't think we should this behavior but it is definitely something you can easily restyle with some custom css.

An option for disabling the icon is not acceptable either, disabling the arrow would bring back the long standing issue of separating a Parent Item that is also a Link and on touch devices it will become impossible to navigate a dropdown.

Hope it all makes sense.

**Comment by alisamii** [comment]
Even if the top item is just a separator and not a link, the icon may not fit in with the design. Custom css can be used to disable the display of the icon, yes, but it should be left to the end user/developer to be able to easily use a one click process in the back end to hide the icon.

You are right that there is an issue with touch devices in this regard, but the solution cannot be a one size fits all, and forcing the icon to be displayed, or more accurately, not providing a simple way to disable that icon cannot be considered best practice.

You are making a stylistic assumption and that's not good.

Best regards,

Ali

**Comment by w00fz** [comment]
I think we can all agree that from the very conception of Menu, a parent menu item has always been represented with text followed by a chevron pointing down. I don't think I have ever seen a parent menu item that wouldn't do that.

Now, I am not saying that hiding the icon is wrong if that's what the design of your template is calling from, but you have to agree that it is kind of specific to your scenario rather than something applicable to everyone else.

Our take for this implementation might not be the best practice but then, what is? The only thing everyone agrees on is that a Parent menu item shouldn't also be a link. This is actually a flaw in Joomla, which allows you to do that but since there are touch devices to take into consideration, it is clear that it can't work anymore just like that. 

We did base this implementation over years of feedbacks we got back from the RocketTheme community while using Gantry 4 and RokNavMenu, this might not be the absolute perfect solution but we think it's definitely the most reliable and flexible we can get and definitely covers all the common case scenarios.

Hiding the dropdown icon is an extra step that is very specific and called by the design, because of that it should be handled by the developer, in my opinion.

**Comment by mahagr** [comment]
If you take a look in any of the modern frameworks like Bootstrap 3, there isn't anymore a concept where you have link and a dropdown in a same block element. And if you have, it works just like ours, except that all the dropdowns need to be clicked before you can see the contents of them.

This is because of in some reports over half of the people are using either phones, tablets or touch screen in their laptop to browse web. Have you ever tried to use Joomla with touch screen? It's a pain...

Most of the frameworks even require you to either click or tap on the menu to open the dropdown, but because of so many people were used to just hover, we decided to change that in our menu.

Basically all this extra functionality makes our menu pretty heavy already and I am recommending making a simpler menu particle if you need the old style menu. Most logic can easily be just thrown away if the mobility features were removed.

**Comment by alisamii** [comment]
Matias and Djamil,

I'm not suggesting throwing away any of the logic. I'm just saying that if you look at many sites, and especially news sites such as CNN BBC or the New York Times, you don't see an arrow or icon next to each menu item name. It just crowds busy menus. So having the ability to hide that icon globally through a setting is a valid request.

Best regards,

Ali

**Comment by mahagr** [comment]
You can easily customize menu not to have icon, but the side effect is that all of those main menu items need to be separators without a link in them. You can easily create your custom menu particle that does just that.
